src/main/kotlin/autolotto/vo/AutoLotto.kt

```diff
@@ -0,0 +1,30 @@
+package autolotto.vo
+
+import kotlin.math.floor
+
+class AutoLotto(val price: Long, private val originLottos: List<Lotto> = listOf()) {
+    val count: Int = initCount(price)
+    val lottos: List<Lotto> = originLottos.ifEmpty { createLottoList() }
+
+    private fun initCount(price: Long): Int {
+        require(price >= 0) { "가격은 음수가 될 수 없습니다." }
+        return (price / LOTTO_PRICE_PER_ONE).toInt()
+    }
+
+    private fun createLottoList(): List<Lotto> {
+        return (1..count).map { Lotto() }
+    }
+
+    fun calculateTotalWinningPrice(winningNumbers: List<Int>): Long {
+        return lottos.map { it.checkWinning(winningNumbers) }.sumOf { it.winningPrice }
+    }
```

Comment:
p3; map 컬렉션 연산을 생략할 수도 있을 거 같아요. 😊

```suggestion
    fun calculateTotalWinningPrice(winningNumbers: List<Int>): Long = lottos.sumOf {
        it.checkWinning(winningNumbers = winningNumbers)
            .winningPrice
    }
```

p2; 승자에 대한 판단을 WinningLotto가 아닌 Lotto가 책임을 가지고 있는 게맞을지 고민을 한 번 해보면 좋을 거 같아요!